plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/widgets/common/MultipleAnnotationHandler.java

```diff
@@ -278,24 +275,59 @@ private static void transferProperties(Target from, Target to, Map<String, Prope
      */
-    private static PropertyTransferPolicy getPolicyForProperty(
+    private static PropertyTransferPolicy getPolicyForAttribute(
         Map<String, PropertyTransferPolicy> policies,
-        String propertyToken) {
+        String value) {
         return policies.entrySet().stream()
-            .filter(entry -> entry.getKey().endsWith(DialogConstants.WILDCARD)
-                ? propertyToken.startsWith(StringUtils.stripEnd(entry.getKey(), DialogConstants.WILDCARD))
-                : propertyToken.equals(entry.getKey()))
+            .filter(entry -> entry.getKey().startsWith(CoreConstants.SEPARATOR_AT))
+            .filter(entry -> {
+                String key = StringUtils.stripStart(entry.getKey(), CoreConstants.SEPARATOR_AT);
+                return key.endsWith(DialogConstants.WILDCARD)
+                    ? value.startsWith(StringUtils.stripEnd(key, DialogConstants.WILDCARD))
+                    : value.equals(key);
+            })
             .map(Map.Entry::getValue)
             .findFirst()
             .orElse(PropertyTransferPolicy.LEAVE_IN_MULTIFIELD);
     }
 
+    /**
+     * Called to pick up an appropriate {@link PropertyTransferPolicy} for a child node from the set of provided policies
+     * @param policies  {@code Map} describing available policies
+     * @param value     {@code Target} object that represents the subnode
+     * @return The selected policy, or the default policy if no appropriate option found
+     */
+    private static PropertyTransferPolicy getPolicyForChildNode(
+        Map<String, PropertyTransferPolicy> policies,
+        Target value) {
+        PropertyTransferPolicy result = Stream.concat(
+            // We put absolute paths before the relative ones so that they are considered in the first turn
+            // even if added late
+            policies.entrySet().stream().filter(entry -> entry.getKey().startsWith(CoreConstants.SEPARATOR_SLASH)),
+            policies.entrySet().stream().filter(entry -> entry.getKey().startsWith(DialogConstants.RELATIVE_PATH_PREFIX))
+        )
+            .filter(entry -> {
+                boolean isAbsolutePath = entry.getKey().startsWith(CoreConstants.SEPARATOR_SLASH);
+                String key = isAbsolutePath ? entry.getKey() : StringUtils.substring(entry.getKey(), DialogConstants.RELATIVE_PATH_PREFIX.length());
+                String checkedValue = isAbsolutePath ? value.getPath() : value.getName();
+                return key.endsWith(DialogConstants.WILDCARD)
+                    ? StringUtils.startsWith(checkedValue, StringUtils.stripEnd(key, DialogConstants.WILDCARD))
+                    : StringUtils.equals(checkedValue, key);
+            })
+            .map(Map.Entry::getValue)
+            .findFirst()
+            .orElse(null);
+        if (result == null) {
+            result = PropertyTransferPolicy.LEAVE_IN_MULTIFIELD;
+        }
+        return result;
+    }
```

Comment:
We can hardly join the two kinds of policy tokens (`/abs/path/...` and `./rel/path/...`) in one filter because we want all absolute paths to be considered before all the relative paths, even if an absolute path was added last. If we join them in one filter, we will then need to employ a comparator, and it will be an extra piece of code anyway.
Concerning the extraction of lengthy lambdas, you are perfectly right